This notebook handles the Phase 1 core training of the FireGuard detection system. To optimize for both speed and accuracy, this pipeline utilizes two key methodologies:

Efficient Data Ingestion via Roboflow. Rather than manually downloading and uploading gigabytes of image data, we utilize the Roboflow API to securely and instantly pull the YOLOv8-formatted D-Fire dataset directly into the cloud environment. This ensures clean, standardized data formatting.

In [2]:
# Install the Ultralytics YOLO framework
!pip install ultralytics

# Import YOLO and check if the NVIDIA GPU is recognized
from ultralytics import YOLO
import torch

print(f"Is CUDA (GPU) available? {torch.cuda.is_available()}")
print(f"GPU Name: {torch.cuda.get_device_name(0) if torch.cuda.is_available() else 'None'}")

Is CUDA (GPU) available? True
GPU Name: Tesla T4


In [3]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [4]:
!pip install roboflow

from roboflow import Roboflow
rf = Roboflow(api_key="4363dTpeKcUXeunhGHJr")
project = rf.workspace("learning-uxvp9").project("d-fire-kymli")
version = project.version(4)
dataset = version.download("yolov8")

loading Roboflow workspace...
loading Roboflow project...


In [5]:
# Load the pre-trained lightweight model
model = YOLO('yolov8n.pt')

# Train the model on the D-Fire dataset
# (We will adjust the 'data' path once your dataset is loaded)
results = model.train(
    data='/content/D-fire-4/data.yaml',
    epochs=30,               # it loops 30 times through the 21,000 images
    imgsz=640,               # Image resolution
    batch=16,                # How many images it processes at once
    project='/content/drive/MyDrive/FireGuard', # Saves directly to your Drive!
    name='lightweight_run'
)

Ultralytics 8.4.36 🚀 Python-3.12.13 torch-2.10.0+cu128 CUDA:0 (Tesla T4, 14913MiB)
engine/trainer: agnostic_nms=False, amp=True, angle=1.0, augment=False, auto_augment=randaugment, batch=16, bgr=0.0, box=7.5, cache=False, cfg=None, classes=None, close_mosaic=10, cls=0.5, compile=False, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=/content/D-fire-4/data.yaml, degrees=0.0, deterministic=True, device=None, dfl=1.5, dnn=False, dropout=0.0, dynamic=False, embed=None, end2end=None, epochs=30, erasing=0.4, exist_ok=False, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, half=False, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=640, int8=False, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=0.01, lrf=0.01, mask_ratio=4, max_det=300, mixup=0.0, mode=train, model=yolov8n.pt, momentum=0.937, mosaic=1.0, multi_scale=0.0, name=lightweight_run, nbs=64, nms=False, opset=None, optimize=False, optimizer=auto, overlap_mask=True, patience=100, per

Full cell output for the training with details:

Downloading https://github.com/ultralytics/assets/releases/download/v8.4.0/yolov8n.pt to 'yolov8n.pt': 100% ━━━━━━━━━━━━ 6.2MB 135.7MB/s 0.0s
Ultralytics 8.4.36 🚀 Python-3.12.13 torch-2.10.0+cu128 CUDA:0 (Tesla T4, 14913MiB)
engine/trainer: agnostic_nms=False, amp=True, angle=1.0, augment=False, auto_augment=randaugment, batch=16, bgr=0.0, box=7.5, cache=False, cfg=None, classes=None, close_mosaic=10, cls=0.5, compile=False, conf=None, copy_paste=0.0, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=/content/D-fire-4/data.yaml, degrees=0.0, deterministic=True, device=None, dfl=1.5, dnn=False, dropout=0.0, dynamic=False, embed=None, end2end=None, epochs=30, erasing=0.4, exist_ok=False, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, half=False, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=640, int8=False, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=0.01, lrf=0.01, mask_ratio=4, max_det=300, mixup=0.0, mode=train, model=yolov8n.pt, momentum=0.937, mosaic=1.0, multi_scale=0.0, name=lightweight_run, nbs=64, nms=False, opset=None, optimize=False, optimizer=auto, overlap_mask=True, patience=100, perspective=0.0, plots=True, pose=12.0, pretrained=True, profile=False, project=/content/drive/MyDrive/FireGuard, rect=False, resume=False, retina_masks=False, rle=1.0, save=True, save_conf=False, save_crop=False, save_dir=/content/drive/MyDrive/FireGuard/lightweight_run, save_frames=False, save_json=False, save_period=-1, save_txt=False, scale=0.5, seed=0, shear=0.0, show=False, show_boxes=True, show_conf=True, show_labels=True, simplify=True, single_cls=False, source=None, split=val, stream_buffer=False, task=detect, time=None, tracker=botsort.yaml, translate=0.1, val=True, verbose=True, vid_stride=1, visualize=False, warmup_bias_lr=0.1, warmup_epochs=3.0, warmup_momentum=0.8, weight_decay=0.0005, workers=8, workspace=None
Downloading https://ultralytics.com/assets/Arial.ttf to '/root/.config/Ultralytics/Arial.ttf': 100% ━━━━━━━━━━━━ 755.1KB 24.7MB/s 0.0s
Overriding model.yaml nc=80 with nc=2

                   from  n    params  module                                       arguments                     
  0                  -1  1       464  ultralytics.nn.modules.conv.Conv             [3, 16, 3, 2]                 
  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]                
  2                  -1  1      7360  ultralytics.nn.modules.block.C2f             [32, 32, 1, True]             
  3                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]                
  4                  -1  2     49664  ultralytics.nn.modules.block.C2f             [64, 64, 2, True]             
  5                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               
  6                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]           
  7                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]              
  8                  -1  1    460288  ultralytics.nn.modules.block.C2f             [256, 256, 1, True]           
  9                  -1  1    164608  ultralytics.nn.modules.block.SPPF            [256, 256, 5]                 
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 12                  -1  1    148224  ultralytics.nn.modules.block.C2f             [384, 128, 1]                 
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 15                  -1  1     37248  ultralytics.nn.modules.block.C2f             [192, 64, 1]                  
 16                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]                
 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 18                  -1  1    123648  ultralytics.nn.modules.block.C2f             [192, 128, 1]                 
 19                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]              
 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 21                  -1  1    493056  ultralytics.nn.modules.block.C2f             [384, 256, 1]                 
 22        [15, 18, 21]  1    751702  ultralytics.nn.modules.head.Detect           [2, 16, None, [64, 128, 256]] 
Model summary: 130 layers, 3,011,238 parameters, 3,011,222 gradients, 8.2 GFLOPs

Transferred 319/355 items from pretrained weights
Freezing layer 'model.22.dfl.conv.weight'
AMP: running Automatic Mixed Precision (AMP) checks...
Downloading https://github.com/ultralytics/assets/releases/download/v8.4.0/yolo26n.pt to 'yolo26n.pt': 100% ━━━━━━━━━━━━ 5.3MB 120.9MB/s 0.0s
AMP: checks passed ✅
train: Fast image access ✅ (ping: 0.0±0.0 ms, read: 850.9±606.4 MB/s, size: 33.4 KB)
train: Scanning /content/D-fire-4/train/labels... 24735 images, 9 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 24735/24735 2.1Kit/s 11.6s
train: New cache created: /content/D-fire-4/train/labels.cache
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, method='weighted_average', num_output_channels=3), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))
val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 356.0±134.4 MB/s, size: 35.3 KB)
val: Scanning /content/D-fire-4/valid/labels... 2285 images, 0 backgrounds, 0 corrupt: 100% ━━━━━━━━━━━━ 2285/2285 1.0Kit/s 2.2s
val: New cache created: /content/D-fire-4/valid/labels.cache
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: MuSGD(lr=0.01, momentum=0.9) with parameter groups 57 weight(decay=0.0), 64 weight(decay=0.0005), 63 bias(decay=0.0)
Plotting labels to /content/drive/MyDrive/FireGuard/lightweight_run/labels.jpg... 
Image sizes 640 train, 640 val
Using 2 dataloader workers
Logging results to /content/drive/MyDrive/FireGuard/lightweight_run
Starting training for 30 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       1/30      2.04G      1.642      2.372      1.617         63        640: 100% ━━━━━━━━━━━━ 1546/1546 2.6it/s 9:56
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 3.5it/s 20.6s
                   all       2285       5194      0.515      0.438       0.44      0.204

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       2/30      2.51G      1.668       1.84      1.622         80        640: 100% ━━━━━━━━━━━━ 1546/1546 2.9it/s 8:47
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 4.0it/s 18.0s
                   all       2285       5194      0.524      0.479      0.482      0.219

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       3/30      2.51G      1.723      1.818      1.675         69        640: 100% ━━━━━━━━━━━━ 1546/1546 3.2it/s 8:09
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 3.7it/s 19.4s
                   all       2285       5194      0.506      0.432      0.431        0.2

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       4/30      2.51G       1.72      1.773      1.683         76        640: 100% ━━━━━━━━━━━━ 1546/1546 3.2it/s 8:08
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 4.1it/s 17.8s
                   all       2285       5194      0.542      0.486      0.498      0.239

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       5/30      2.51G      1.664      1.665      1.635         49        640: 100% ━━━━━━━━━━━━ 1546/1546 3.1it/s 8:12
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 4.0it/s 17.9s
                   all       2285       5194      0.573      0.521      0.544      0.269

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       6/30      2.51G      1.629      1.585      1.607         66        640: 100% ━━━━━━━━━━━━ 1546/1546 3.2it/s 8:04
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 3.9it/s 18.3s
                   all       2285       5194      0.617      0.563      0.599      0.307

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       7/30      2.51G      1.596      1.528      1.586         69        640: 100% ━━━━━━━━━━━━ 1546/1546 3.1it/s 8:12
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 4.0it/s 18.0s
                   all       2285       5194       0.64      0.584      0.628      0.323

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       8/30      2.51G      1.578      1.485      1.568         59        640: 100% ━━━━━━━━━━━━ 1546/1546 3.1it/s 8:12
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 4.0it/s 18.0s
                   all       2285       5194      0.631      0.595      0.633      0.331

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
       9/30      2.51G      1.556      1.444      1.548         49        640: 100% ━━━━━━━━━━━━ 1546/1546 3.2it/s 8:09
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 3.7it/s 19.3s
                   all       2285       5194      0.654      0.619       0.66      0.354

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      10/30      2.51G      1.542       1.42      1.539         60        640: 100% ━━━━━━━━━━━━ 1546/1546 3.1it/s 8:13
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 3.9it/s 18.3s
                   all       2285       5194      0.671      0.638      0.672      0.356

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      11/30      2.51G      1.523      1.385      1.524         63        640: 100% ━━━━━━━━━━━━ 1546/1546 3.1it/s 8:18
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 4.0it/s 17.8s
                   all       2285       5194      0.677      0.625      0.683      0.363

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      12/30      2.51G      1.506      1.353      1.513         63        640: 100% ━━━━━━━━━━━━ 1546/1546 3.1it/s 8:13
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 4.0it/s 17.9s
                   all       2285       5194      0.667      0.641      0.684      0.368

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      13/30      2.51G      1.502      1.331      1.501         63        640: 100% ━━━━━━━━━━━━ 1546/1546 3.2it/s 8:04
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 4.1it/s 17.7s
                   all       2285       5194      0.702      0.638      0.699      0.382

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      14/30      2.51G      1.476      1.314      1.488         51        640: 100% ━━━━━━━━━━━━ 1546/1546 3.1it/s 8:11
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 3.9it/s 18.4s
                   all       2285       5194      0.701      0.657      0.709       0.39

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      15/30      2.51G      1.477        1.3      1.485         57        640: 100% ━━━━━━━━━━━━ 1546/1546 3.2it/s 8:07
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 4.0it/s 18.0s
                   all       2285       5194      0.699      0.659      0.714      0.395

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      16/30      2.51G       1.46      1.268       1.47         57        640: 100% ━━━━━━━━━━━━ 1546/1546 3.2it/s 8:04
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 4.1it/s 17.7s
                   all       2285       5194      0.712      0.667      0.718      0.396

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      17/30      2.51G       1.45      1.252      1.461         54        640: 100% ━━━━━━━━━━━━ 1546/1546 3.2it/s 8:03
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 4.1it/s 17.6s
                   all       2285       5194      0.715      0.666      0.721      0.397

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      18/30      2.51G      1.443      1.238      1.455         57        640: 100% ━━━━━━━━━━━━ 1546/1546 3.2it/s 8:05
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 4.0it/s 18.1s
                   all       2285       5194      0.718      0.664      0.725      0.402

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      19/30      2.51G      1.427      1.215      1.448         49        640: 100% ━━━━━━━━━━━━ 1546/1546 3.1it/s 8:11
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 4.0it/s 17.8s
                   all       2285       5194      0.715       0.67      0.726      0.404

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      20/30      2.51G      1.415      1.198      1.434         64        640: 100% ━━━━━━━━━━━━ 1546/1546 3.2it/s 8:11
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 3.8it/s 18.8s
                   all       2285       5194      0.721      0.684      0.737      0.411
Closing dataloader mosaic
albumentations: Blur(p=0.01, blur_limit=(3, 7)), MedianBlur(p=0.01, blur_limit=(3, 7)), ToGray(p=0.01, method='weighted_average', num_output_channels=3), CLAHE(p=0.01, clip_limit=(1.0, 4.0), tile_grid_size=(8, 8))

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      21/30      2.51G      1.495      1.139      1.504         20        640: 100% ━━━━━━━━━━━━ 1546/1546 3.2it/s 7:60
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 3.7it/s 19.5s
                   all       2285       5194      0.728      0.679       0.73       0.41

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      22/30      2.51G      1.475      1.098      1.487         32        640: 100% ━━━━━━━━━━━━ 1546/1546 3.3it/s 7:53
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 4.0it/s 18.0s
                   all       2285       5194      0.724      0.691      0.739      0.414

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      23/30      2.51G      1.454       1.07       1.47         30        640: 100% ━━━━━━━━━━━━ 1546/1546 3.2it/s 7:58
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 3.7it/s 19.4s
                   all       2285       5194      0.731      0.686      0.739      0.413

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      24/30      2.51G      1.434      1.041      1.456         23        640: 100% ━━━━━━━━━━━━ 1546/1546 3.2it/s 7:56
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 4.0it/s 18.0s
                   all       2285       5194      0.733       0.69       0.74      0.416

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      25/30      2.51G       1.42      1.016      1.444         23        640: 100% ━━━━━━━━━━━━ 1546/1546 3.2it/s 8:00
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 4.0it/s 18.0s
                   all       2285       5194      0.721      0.696       0.74      0.416

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      26/30      2.51G      1.401     0.9899      1.428         27        640: 100% ━━━━━━━━━━━━ 1546/1546 3.2it/s 7:57
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 3.8it/s 19.0s
                   all       2285       5194      0.739       0.69      0.742      0.416

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      27/30      2.51G       1.38     0.9687       1.41         36        640: 100% ━━━━━━━━━━━━ 1546/1546 3.2it/s 7:56
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 3.7it/s 19.3s
                   all       2285       5194      0.728      0.707      0.745      0.418

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      28/30      2.51G      1.362      0.943      1.397         29        640: 100% ━━━━━━━━━━━━ 1546/1546 3.2it/s 8:02
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 3.9it/s 18.3s
                   all       2285       5194       0.73      0.705      0.744      0.418

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      29/30      2.51G       1.35     0.9264      1.388         31        640: 100% ━━━━━━━━━━━━ 1546/1546 3.3it/s 7:55
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 3.8it/s 19.1s
                   all       2285       5194      0.729      0.707      0.744      0.417

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      30/30      2.51G      1.337     0.9077      1.374         30        640: 100% ━━━━━━━━━━━━ 1546/1546 3.3it/s 7:53
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 4.1it/s 17.6s
                   all       2285       5194      0.734      0.705      0.746      0.418

30 epochs completed in 4.241 hours.
Optimizer stripped from /content/drive/MyDrive/FireGuard/lightweight_run/weights/last.pt, 6.2MB
Optimizer stripped from /content/drive/MyDrive/FireGuard/lightweight_run/weights/best.pt, 6.2MB

Validating /content/drive/MyDrive/FireGuard/lightweight_run/weights/best.pt...
Ultralytics 8.4.36 🚀 Python-3.12.13 torch-2.10.0+cu128 CUDA:0 (Tesla T4, 14913MiB)
Model summary (fused): 73 layers, 3,006,038 parameters, 0 gradients, 8.1 GFLOPs
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% ━━━━━━━━━━━━ 72/72 3.4it/s 21.4s
                   all       2285       5194      0.732      0.706      0.746      0.418
                  fire       1127       2853      0.711      0.629       0.69      0.356
                 smoke       2072       2341      0.752      0.783      0.801      0.481
Speed: 0.2ms preprocess, 2.1ms inference, 0.0ms loss, 2.2ms postprocess per image
Results saved to /content/drive/MyDrive/FireGuard/lightweight_run
